packages/front-end/components/Metrics/DateGraph.tsx

```diff
@@ -107,29 +121,94 @@ const DateGraph: FC<{
             d: row.key,
             v: row.total / row.users,
             s: row.stddev,
+            u: row.users,
           };
         }),
     [dates, groupby]
   );
 
+  const getTooltipData = (
+    mx: number
+  ): { x: number; y: number; d: Datapoint } => {
+    const innerWidth = width - margin[1] - margin[3] + width / data.length - 1;
+    const px = mx / innerWidth;
+    const index = Math.max(
+      Math.min(Math.round(px * data.length), data.length - 1),
+      0
+    );
+    const d = data[index];
+    const x = (data.length > 0 ? index / data.length : 0) * innerWidth;
+    const y = yScale(d.v) ?? 0;
+    return { x, y, d };
+  };
+  const getTooltipContents = ({ d }) => {
+    return (
+      <>
+        <div className={styles.val}>
+          {formatConversionRate(type, d.v as number)}
+        </div>
+        {"s" in d && (
+          <div className={styles.secondary}>&sigma;: {d.s.toFixed(2)}</div>
+        )}
+        {"u" in d && <div className={styles.secondary}>n: {d.u}</div>}
+        <div className={styles.date}>{date(d.d as Date)}</div>
+      </>
+    );
+  };
+
+  const [tooltipShouldDetectBounds] = useState(true);
+
+  const { containerRef, containerBounds } = useTooltipInPortal({
+    scroll: true,
+    detectBounds: tooltipShouldDetectBounds,
+  });
+
+  const {
+    showTooltip,
+    hideTooltip,
+    tooltipOpen,
+    tooltipData,
+    tooltipLeft = 0,
+    tooltipTop = 0,
+  } = useTooltip<TooltipData>();
+
+  const handlePointer = useCallback(
+    (event: React.PointerEvent<HTMLDivElement>) => {
+      // coordinates should be relative to the container in which Tooltip is rendered
+      const containerX =
+        ("clientX" in event ? event.clientX : 0) - containerBounds.left;
+      const data = getTooltipData(containerX);
+      showTooltip({
+        tooltipLeft: data.x,
+        tooltipTop: data.y,
+        tooltipData: data,
+      });
+    },
+    [showTooltip, tooltipShouldDetectBounds, containerBounds, data]
+  );
+
+  let xScale = null;
```

Comment:
needed `xScale`, `yScale`, and `width` available outside the render loop. `width` is tricky since it's a side effect of initial render, but not sure how else to deal with it.

Reply:
I think you can avoid this by
1.  Having `getTooltipData` accept width and yscale as arguments
2.  Defining `handlePointer` inside the `<ParentSizeModern>` component without using `useCallback`.  If performance is an issue, we can wrap the stuff inside `<ParentSizeModern>` in a new child component that takes width, etc. as props.

Reply:
Ok, inlined `handlePointer` inside `<ParentSizeModern>`. Kind of funky since this runs every render update, but seems to run without any issues for me. Maybe we can come back to this later and make a HOC or child component.